internal/provider/common_cert.go

```diff
@@ -448,39 +449,52 @@ func updateCertificate(_ context.Context, _ *schema.ResourceData, _ interface{})
 	return nil
 }
 
-// distinguishedNamesFromSubjectAttributes it takes a map subject attributes and
-// converts it to a pkix.Name (X.509 distinguished names).
-func distinguishedNamesFromSubjectAttributes(nameMap map[string]interface{}) *pkix.Name {
+// getSubjectDistinguishedNames it looks for the "subject" attribute: if found and non-empty, it returns
+// a *pkix.Name useful for creating x509.Certificate or x509.CertificateRequest.
+func getSubjectDistinguishedNames(d *schema.ResourceData) *pkix.Name {
```

Comment:
Totally a small nit, but something I've found super helpful for potentially adding unit testing (and maybe helping with the framework migration) is creating functions which use the attributes value type (in this case a `[]interface{}` parameter) where the calling code does the necessary work to get data out of `ResourceData` (in this case `getSubjectDistinguishedNames(d.Get("subject").([]interface{}))`).